mqtt/mqtt-edgehub/tests/authorization.rs

```diff
@@ -0,0 +1,105 @@
+use futures_util::StreamExt;
+
+use mqtt3::{
+    proto::ClientId, proto::PacketIdentifier, proto::QoS, proto::Subscribe, proto::SubscribeTo,
+};
+use mqtt_broker::{auth::DenyAll, BrokerBuilder};
+use mqtt_broker_tests_util::{
+    client::TestClientBuilder,
+    packet_stream::PacketStream,
+    server::{start_server, DummyAuthenticator},
+};
+use mqtt_edgehub::{
+    auth::EdgeHubAuthorizer, auth::IdentityUpdate, command::AuthorizedIdentitiesCommand,
+};
+
+mod common;
+
+/// Scenario:
+// create broker
+// create command handler
+// publish authorization update
+// connect authorized client and subscribe
+// publish authorization update with client removed
+// verify client has disconnected
+#[tokio::test]
+async fn disconnect_client_on_auth_update() {
+    // Start broker with DummyAuthorizer that allows everything from CommandHandler and $edgeHub,
+    // but otherwise passes authorization along to EdgeHubAuthorizer
+    let broker = BrokerBuilder::default()
+        .with_authorizer(common::DummyAuthorizer::new(EdgeHubAuthorizer::new(
+            DenyAll,
+        )))
+        .build();
+    let broker_handle = broker.handle();
+
+    let server_handle = start_server(broker, DummyAuthenticator::anonymous());
+
+    // start command handler with AuthorizedIdentitiesCommand
+    let command = AuthorizedIdentitiesCommand::new(&broker_handle);
+    let (command_handler_shutdown_handle, join_handle) =
+        common::start_command_handler(common::TEST_SERVER_ADDRESS.to_string(), command)
+            .await
+            .expect("could not start command handler");
+
+    let mut edgehub_client = TestClientBuilder::new(server_handle.address())
+        .with_client_id(ClientId::IdWithCleanSession("$edgehub".into()))
+        .build();
+
+    let service_identity1 =
+        IdentityUpdate::new("device-1".into(), Some("device-1;$edgehub".into()));
+    let identities: Vec<IdentityUpdate> = vec![service_identity1];
```

Comment:
nit:
```suggestion
    let identities = vec![service_identity1];
```

Reply:
I have a hard time letting go of my explicit types